src/unitxt/metrics.py

```diff
@@ -317,6 +317,318 @@ def update_and_adjust_global_score(
                 instance["score"]["global"].pop(score_ci)
 
 
+def new_random_generator():
+    # The np.random.default_rng expects a 32-bit int, while hash(..) can return a 64-bit integer.
+    # So use '& MAX_32BIT' to get a 32-bit seed.
+    _max_32bit = 2**32 - 1
+    return np.random.default_rng(hash(get_seed()) & _max_32bit)
+
+
+class ConfidenceIntervalMixin(Artifact):
+    n_resamples: int = 1000
+    confidence_level: float = 0.95
+    ci_score_names: List[str] = None
+
+    @abstractmethod
+    def _sample_to_scores(self, sample: List[Any]) -> Dict[str, Any]:
+        pass
+
```

Comment:
to my understanding, score_names might include instance metrics as well as global metric.
For instance metric, the instance scores are already computed, and why compute them again?